examples/lighting-app/esp32/README.md

```diff
@@ -111,6 +111,33 @@ make sure the IDF_PATH has been exported(See the manual setup steps above).
 
         $ idf.py -p /dev/tty.SLAB_USBtoUART monitor
 
+## Generating and flashing chip-factory NVS binary image
+
+If `CONFIG_ENABLE_ESP32_COMMISSIONABLE_DATA_PROVIDER` option is enabled then
+ESP32 specific commissionable data provider is used, otherwise the default/test
+commissionable data provider in GenericConfigurationManagerImpl is used.
+
+Config option `CONFIG_CHIP_FACTORY_NAMESPACE_PARTITION_LABEL` lets us configure
+the partition name for reading `chip-factory` data. Default is `nvs`.
+
+If `CONFIG_ENABLE_ESP32_COMMISSIONABLE_DATA_PROVIDER` options is enabled then
+chip-factory data must be flashed into the configure partition.
+
+`scripts/tools/generate_esp32_chip_factory_bin.py` script generates the
+chip-factory NVS image `partition.bin`. It takes three arguments, passcode,
+discriminator and partition size(optional, default:0x6000).
```

Comment:
Discriminator and verifier take < 128 bytes togets. Why would you allocate a 0x6000 partition for this? I would also recommend illustrating a `DeviceAttestationCredentialsProvider` (e.g. `ESP32DeviceAttestationCredentialsProvider` that makes use of factory partition storage of DAC, PAI, CD, and safely stores attestation key. The same single class could implement both `CommissionableDataProvider` and `DeviceAttestationCredentialsProvider` interfaces and re-use some utilities to access the NVS partition...

Reply:
For generating an NVS partition, minimum NVS Partition Size needed is 0x3000 bytes but, I took 0x6000 from the [partitions.csv](https://github.com/project-chip/connectedhomeip/blob/master/examples/lighting-app/esp32/partitions.csv#L3) present in esp32 apps. Space after discriminator and spake2p params will be empty and can be used by application.

I am going to add a follow-up PR for DAC Provider.

Reply:
I have implemented your suggestion, `ESP32FactoryDataProvider` which implements `DeviceAttestationCredentialsProvider` and `CommissionableDataProvider`, please take a look.